Description: Port Py_InitModule* calls to PyModule_Create for python3.
Author: Adam Conrad <adconrad@ubuntu.com>
Forwarded: no

--- twisted-12.3.0.orig/twisted/python/_initgroups.c
+++ twisted-12.3.0/twisted/python/_initgroups.c
@@ -58,9 +58,27 @@ static PyMethodDef InitgroupsMethods[] =
 
 #endif /* defined(__unix__) || defined(unix) */
 
+#if PY_MAJOR_VERSION >= 3
+static struct PyModuleDef moduledef = {
+	PyModuleDef_HEAD_INIT,
+	"_initgroups",       /* m_name */
+	NULL,                /* m_doc */
+	-1,                  /* m_size */
+	InitgroupsMethods,   /* m_methods */
+	NULL,                /* m_reload */
+	NULL,                /* m_traverse */
+	NULL,                /* m_clear */
+	NULL,                /* m_free */
+};
+#endif
+
 void
 init_initgroups(void)
 {
+#if PY_MAJOR_VERSION >= 3
+	PyModule_Create(&moduledef);
+#else
 	Py_InitModule("_initgroups", InitgroupsMethods);
+#endif
 }
 
--- twisted-12.3.0.orig/twisted/python/sendmsg.c
+++ twisted-12.3.0/twisted/python/sendmsg.c
@@ -116,6 +116,19 @@ static PyMethodDef sendmsg_methods[] = {
     {NULL, NULL, 0, NULL}
 };
 
+#if PY_MAJOR_VERSION >= 3
+static struct PyModuleDef moduledef = {
+    PyModuleDef_HEAD_INIT,
+    "sendmsg",           /* m_name */
+    sendmsg_doc,         /* m_doc */
+    -1,                  /* m_size */
+    sendmsg_methods,     /* m_methods */
+    NULL,                /* m_reload */
+    NULL,                /* m_traverse */
+    NULL,                /* m_clear */
+    NULL,                /* m_free */
+};
+#endif
 
 PyMODINIT_FUNC initsendmsg(void) {
     PyObject *module;
@@ -123,7 +136,11 @@ PyMODINIT_FUNC initsendmsg(void) {
     sendmsg_socket_error = NULL; /* Make sure that this has a known value
                                     before doing anything that might exit. */
 
+#if PY_MAJOR_VERSION >= 3
+    module = PyModule_Create(&moduledef);
+#else
     module = Py_InitModule3("sendmsg", sendmsg_methods, sendmsg_doc);
+#endif
 
     if (!module) {
         return;
